docs/reference/query-dsl/queries/mlt-query.asciidoc

```diff
@@ -81,89 +85,155 @@ this case should be preferred over only using `like_text`.
-`per_field_analyzer` parameter of the <<docs-termvectors-per-field-analyzer,Term Vectors API>>.
-
-|`include` | When using `ids` or `docs`, specifies whether the documents should be
-included from the search. Defaults to `false`.
-
-|`minimum_should_match`| coming[1.5.0] From the generated query, the number of terms that
-must match following the <<query-dsl-minimum-should-match,minimum should
-syntax>>. (Defaults to `"30%"`).
-
-|`percent_terms_to_match` | deprecated[1.5.0,Replaced by `minimum_should_match`]
-From the generated query, the percentage of terms that must match (float value
-between 0 and 1). Defaults to `0.3` (30 percent).
-
-|`min_term_freq` |The frequency below which terms will be ignored in the
-source doc. The default frequency is `2`.
-
-|`max_query_terms` |The maximum number of query terms that will be
-included in any generated query. Defaults to `25`.
-
-|`stop_words` |An array of stop words. Any word in this set is
-considered "uninteresting" and ignored. Even if your Analyzer allows
-stopwords, you might want to tell the MoreLikeThis code to ignore them,
-as for the purposes of document similarity it seems reasonable to assume
-that "a stop word is never interesting".
-
-|`min_doc_freq` |The frequency at which words will be ignored which do
-not occur in at least this many docs. Defaults to `5`.
-
-|`max_doc_freq` |The maximum frequency in which words may still appear.
-Words that appear in more than this many docs will be ignored. Defaults
-to unbounded.
+==== How it Works
+
+Suppose we wanted to find all documents similar to a given input document.
+Obviously, the input document itself should be its best match for that type of
+query. And the reason would be mostly, according to
+link:https://lucene.apache.org/core/4_9_0/core/org/apache/lucene/search/similarities/TFIDFSimilarity.html[Lucene scoring formula],
+due to the terms with the highest tf-idf. Therefore, the terms of the input
+document that have the highest tf-idf are good representatives of that
+document, and could be used within a disjunctive query (or `OR`) to retrieve
+similar documents. The MLT query simply extracts the text from the input
+document, analyzes it, usually using the same analyzer at the field, then
+selects the top K terms with highest tf-idf to form a disjunctive query of
+these terms.
+
+IMPORTANT: The fields on which to perform MLT must be indexed and of type
+`string`. Additionally, when using `like` with documents, either `_source`
+must be enabled or the fields must be `stored` or store `term_vector`. In
+order to speed up analysis, it could help to store term vectors at index time.
+
+For example, if we wish to perform MLT on the "title" and "tags.raw" fields,
+we can explicitly store their `term_vector` at index time. We can still
+perform MLT on the "description" and "tags" fields, as `_source` is enabled by
+default, but there will be no speed up on analysis for these fields.
```

Comment:
Indeed it would be nice to know the tipping point. What takes time is re-tokenizing the text and calling docFreq() on every term, as well as executing the query itself. In my test, when asking for 25 terms only, and with documents less 5 KB, the execution speed is dominated by the query itself however. But this could very well be different with larger documents.